imbi/user.py

```diff
@@ -171,6 +172,13 @@ def has_permission(self, value: str) -> bool:
     def on_postgres_error(_metric_name: str, exc: Exception) -> None:
         raise web.HTTPError(500, 'System error')
 
+    @classmethod
+    def on_token_postgres_error(cls, metric_name: str, exc: Exception) -> None:
+        """Pass InvalidTextRepresentation exceptions through."""
+        if isinstance(exc, psycopg2.errors.InvalidTextRepresentation):
+            raise exc
+        return cls.on_postgres_error(metric_name, exc)
```

Comment:
I've always used `super().` instead of `cls.` is there a benefit in not using `super()` here?